### ISSUE TITLE
After updating from Formie 1 to 2, the FileMaker Webhook isn't sending assets' URLs anymore

### ISSUE THREAD

**Issue by romainpoirier** [verbb/formie#1520]
### Describe the bug

After updating from Formie 1 to 2, the FileMaker Webhook isn't sending assets' URLs anymore, while he asset's filesystem have public URLs.

This is an example of data that got from the payload test for the file's input:
```
"file": [
  {
    "isFolder": false,
    "sourcePath": null,
    "folderId": 3865,
    "uploaderId": null,
    "folderPath": "332738/",
    "kind": "image",
    "alt": null,
    "size": 95989,
    "keptFile": null,
    "dateModified": {
      "date": "2023-08-03 16:24:41.000000",
      "timezone_type": 3,
      "timezone": "Europe/Brussels"
    },
    "newLocation": null,
    "locationError": null,
    "newFilename": null,
    "newFolderId": null,
    "tempFilePath": null,
    "avoidFilenameConflicts": false,
    "suggestedFilename": null,
    "conflictingFilename": null,
    "deletedWithVolume": false,
    "keepFileOnDelete": false,
    "id": 332739,
    "tempId": null,
    "draftId": null,
    "revisionId": null,
    "isProvisionalDraft": false,
    "uid": "d63cbdc9-94b1-431a-ad54-d5539bac8886",
    "siteSettingsId": 402882,
    "fieldLayoutId": 89,
    "structureId": null,
    "contentId": 221304,
    "enabled": true,
    "archived": false,
    "siteId": 1,
    "title": "Maria B Foto color2 500x500",
    "slug": null,
    "uri": null,
    "dateCreated": {
      "date": "2023-08-03 16:24:41.000000",
      "timezone_type": 3,
      "timezone": "Europe/Brussels"
    },
    "dateUpdated": {
      "date": "2023-08-03 16:24:41.000000",
      "timezone_type": 3,
      "timezone": "Europe/Brussels"
    },
    "dateLastMerged": null,
    "dateDeleted": null,
    "root": null,
    "lft": null,
    "rgt": null,
    "level": null,
    "searchScore": null,
    "trashed": false,
    "awaitingFieldValues": false,
    "propagating": false,
    "propagateAll": false,
    "newSiteIds": [],
    "isNewForSite": false,
    "resaving": false,
    "duplicateOf": null,
    "firstSave": false,
    "mergingCanonicalChanges": false,
    "updatingFromDerivative": false,
    "previewing": false,
    "hardDelete": false
  }
],
```

### Steps to reproduce

1. Setup the integration from the form: Integrations > Webhooks > FileMaker Webhook > Enabled
2. Setup a File Upload field on the from
3. Check the payload received in the target URL

### Form settings

- Multi-page form: Yes
- Submission Method: Page Reload
- Client-side Validation: Yes
- Custom Form Templates: No

### Craft CMS version

4.4.14

### Plugin version

2.0.32

### Multi-site?

Yes

**Comment by engram-design** [comment]
So the FileMaker Webhook must be a custom integration you've written. Just wanted to confirm if it's all inline with the Formie [Webhook](https://github.com/verbb/formie/blob/craft-4/src/integrations/webhooks/Webhook.php) integration, in particular how the payload sent to the webhook endpoint is constructed? Just wanted to ensure you're not doing anything custom that might produce a different result.

I'll double check this, but this should be the serialized data of an Asset element, which is Craft's responsibility. I would find it strange there is no URL for an asset that is publicly visible.

**Comment by engram-design** [comment]
Actually, as soon as I wrote that, I knew what the issue might be. It'll be due to how we serialize elements for webhook payloads.

Updated for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.0.33"`.

**Comment by romainpoirier** [comment]
Let's focus on the data sent from the Formie Webhook integration, that's where the problem is. The file is embeded in the payload as an object of data, but not the file URL and name.

Here's an example is what the integration is sending (where `squareStill` is the field handle):
```
"squareStill": [{
	"isFolder": false,
	"sourcePath": NULL,
	"folderId": 3810,
	"folderPath": "forms/loremIpsum/326461/326985562/",
	"kind": "image",
	"alt": NULL,
	"size": 478116
	"keptFile": NULL
	"newLocation": NULL,
	"locationError": NULL,
	"newFilename": NULL,
	"newFolderId": NULL,
	"tempFilePath": NULL,
	"avoidFilenameConflicts": false,
	"suggestedFilename": NULL,
	"conflictingFilename": NULL,
	"deletedWithVolume": false,
	"keepFileOnDelete": false,
	"id": 326906,
	"tempId": NULL,
	"draftId": NULL,
	"revisionId": NULL,
	"uploaderId": 326461,
	"fieldLayoutId": 3,
	"structureId": NULL,
	"contentId": 218285
	"enabled": true,
	"archived": false,
	"siteId": 1,
	"title": "Bads Agents 2 Square Image",
	"slug": NULL,
	"uri": NULL,
	...
}]
```

However, when I export the submission to JSON using Craft's CP native export button, the file is embedded as an URL (where `1 square still` is the field name, that have the handle `squareStill`):
`"1 square still": "https://www.example.com/assets/files/forms/loremIpsum/326461/326985562/cat.jpg",`

**Comment by romainpoirier** [comment]
Thank you for the `"dev-craft-4 as 2.0.33"`. I can confirm that this fix is working.

**Comment by engram-design** [comment]
Fixed in [2.0.34](https://github.com/verbb/formie/releases/tag/2.0.34)
